disnake/ext/commands/base_core.py

```diff
@@ -132,6 +133,8 @@ class InvokableApplicationCommand(ABC):
     body: ApplicationCommand
 
     def __init__(self, func: CommandCallback, *, name: str = None, **kwargs):
+        self.__original_kwargs__ = kwargs.copy()
```

Comment:
Similarly to `Command`, this should be copied in `__new__` instead in case any subtypes modify the kwargs before calling their `super.__init__` (right now there are none, but it would prevent possible future issues)

I believe doing that should also help avoid having to re-implement `_ensure_assignment_on_copy` and `copy` in all subtypes

Reply:
Moved `__original_kwargs__` to `__new__` for now, but leaving this open as I'm not quite sure how to go about the second part of this comment. I shall think about it a bit in the meantime.